### DISCUSSION TITLE
Support for emergency/crisis situations

### DISCUSSION THREAD

**Discussion by crodriguezvega** [cosmos/ibc-go#1351] [Ideas]
With the recent events around Terra and the risk of of governance attacks following the severe devaluation of Luna, the Terra team decided to [close channels to Osmosis and Crescent](https://github.com/terra-money/core/pull/761/files). There is currently no mechanisms in ibc-go to facilitate actions like this in emergency circumstances.

Opening this discussion to collect ideas of what ibc-go can/should do to support developers/users under crisis situations.

**Comment by seantking** [comment]
Some questions different users may have:

1. As a cosmos SDK developer/team building a chain what is the best way for me to disable channels rapidly in case of emergencies for currently existing IBC applications (ics20, ics27, ics29)?

2. As an IBC application developer what considerations/assumptions if any do I need to make with regards to the closing of channels in emergency situations when building an IBC app (for downstream consumers/teams using the module)?

3. How much of this should be handled by core IBC (with regards to potentially improving UX) & and how much of this is left as responsibility for IBC application developers/downstream users i.e. SDK devs)?
